tests/periph_rtt_min/main.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test to figure out RTT_MIN_OFFSET
+ *
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdatomic.h>
+
+#include "cpu.h"
+#include "periph_conf.h"
+#include "periph/rtt.h"
+
+#include "xtimer.h"
+
+/* min. amount of time to wait between set_alarm() */
+#define MIN_WAIT_US     (100UL)
+#define US_PER_TICK     (US_PER_SEC / RTT_FREQUENCY)
+
+#ifndef SAMPLES
+#define SAMPLES 1024LU
+#endif
+
+static mutex_t _mtx = MUTEX_INIT_LOCKED;
+
+void cb(void *arg)
+{
+    (void)arg;
+    mutex_unlock(&_mtx);
+}
+
+int main(void)
+{
+    uint32_t value = 0;
+
+    rtt_init();
+
+    printf("Evaluate RTT_MIN_OFFSET over %" PRIu32 " samples\n",
+           (uint32_t)SAMPLES);
+
+    for (unsigned i = 0; i < SAMPLES; i++) {
+        uint32_t offset = 0;
+        int ret = -1;
+        while (ret != 0) {
+            offset++;
+            uint32_t now = rtt_get_counter();
+            mutex_trylock(&_mtx);
```

Comment:
What is that for?

Reply:
Since there are multiple iterations you always want the mutex to be locked before `xtimer_mutex_lock_timeout`, if it timed out then it would not be locked on each new "iteration". I added a comment.

Reply:
> if it timed out then it would not be locked on each new "iteration"

If `xtimer_mutex_lock_timeout()` times out the mutex is still locked.

Reply:
True I'll remove.